cogs/ahk/ahk.py

```diff
@@ -743,7 +743,34 @@ async def solved(self, inter: disnake.AppCmdInter):
 
         await inter.channel.edit(
             archived=True,
-            applied_tags=inter.channel.applied_tags + [solved_tag],
+            applied_tags=inter.channel.applied_tags + ([] if solved_tag in inter.channel.applied_tags else [solved_tag]),
```

Comment:
Figure out the new applied_tags array outside of this method call, something like

```py
add = dict()
if solved_tag not in inter.channel.applied_tags:
    add["applied_tags"] = inter.channel.applied_tags + [solved_tag]
    
await inter.channel.edit(
    archived=True,
    **add,
)
```